sky/serve/serve_utils.py

```diff
@@ -242,9 +251,20 @@ def set_service_status_from_replica_statuses(
         # terminated, the service status will still be READY, but we don't want
         # change service status to READY.
         return
-    serve_state.set_service_status(
+
+    ready_replicas = list(filter(lambda info: info.is_ready, replica_infos))
+    if update_mode == UpdateMode.ROLLING:
+        active_versions = sorted(
+            list(set(info.version for info in ready_replicas)))
+    else:
+        chosen_version = get_latest_version_with_min_replicas(
+            service_name, replica_infos)
+        active_versions = [chosen_version] if chosen_version is not None else []
```

Comment:
wonder why is list used here (can it be `active_version`?), it seems `active_versions` can only contain 0 or 1 version?

Reply:
It can be multiple versions if we are using rolling update, since the load balancer can spread across multiple versions. We use a list here to keep the API consistent : )